### PR TITLE
Update TFC variables page

### PR DESCRIPTION
**Pull request by laurapacilio**
This PR closes https://github.com/hashicorp/terraform-website/issues/1829.
This PR is blocked by: https://github.com/GoogleCloudPlatform/magic-modules/pull/5228

The user asked for examples of the environment variables that you need to set for each major cloud provider in TFC. I believe this is because they were having a hard time finding this information in the provider docs. After some thought, I decided not to add the examples into our documentation because a) we won't do that for every provider and b) the information is already documented with examples in the Terraform Registry documentation for each provider. However, I do think a slightly more detailed explanation of environment variables is warranted, and perhaps we can point users to the correct place in the registry documentation for major providers. 

This information also didn't seem like it naturally fit anywhere in the current page structure. So instead I:
- Added a new "Variable Types" section into the page where I explain both input variables and environment variables a bit more thoroughly and direct users to consult the registry documentation for provider environment variables, pointing to the docs for Azure, GCP, and AWS as an example.
- Added a callout to a learn tutorial where we set both input variables and environment variables as credentials. I think this provides a great example so folks can see how it's done, links to learn, AND doesn't repeat existing information.
- Reworked the introduction to hopefully help it flow better, add an opening sentence explaining why variables are useful, and also to incorporate the note at the top into the main text. We should reserve notes for times when we really need to call the user's attention to something.

I have:
- [x] Verified that this previews successfully on my local machine. 
- [x] Added at least one tag.
- [x] Requested at least one reviewer.

### CODE REVIEW

**Comment by laurapacilio** [comment]
Please note that the link checker is flagging missing anchors in linked provider docs...but in local testing these anchors worked perfectly fine so I think we can ignore. Thank you!

File: content/source/docs/cloud/workspaces/variables.html.md
Comment (laurapacilio) [reply content/source/docs/cloud/workspaces/variables.html.md:12]: good to know!! thank you!

**Review by judithpatudith — APPROVED** [review]
Hey Laura, thanks for this, it's awesome. I really like linking out to the big three provider docs to show examples of variables you can set. 

The one thing I'm wondering about are cases like the GCP credentials. The documented process for giving Terraform access to your credentials doesn't work on Terraform Cloud unless you put your credentials file onto the Terraform worker machine (which we aren't really providing instructions for in the provider docs). So I think while this is a nice way to provider users an example of common available Environment Variables, it doesn't exactly fix the user's problem for all providers. 

We might want to add to the GCP documentation to show users what to do (maybe there's a way to flatten the JSON and set it directly)? 

Gonna pre-approve because I don't want to block the PR, but we might want to remove the GCP link until we add a documented way to put GCP creds into TFC? I may be naive and just missing how to do it?

File: content/source/docs/cloud/workspaces/variables.html.md
Comment (im2nguyen) [inline content/source/docs/cloud/workspaces/variables.html.md:23]: @judithpatudith brought up a good point that I completely missed reviewing!

After digging into the GCP docs a little bit more, we should link them to the [GCP configuration reference docs](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference#full-reference) instead. You can authenticate to the GCP provider on TFC using the `GOOGLE_CREDENTIALS` environment variables. This [discuss post](https://discuss.hashicorp.com/t/how-to-authenticate-to-gcp-when-using-terraform-cloud/17489/9) dives into more detail.

If possible, we should create a PR to the GCP provider to make this more clear tho.

```suggestion
**Environment variables** can store provider credentials and other data. Refer to your provider's Terraform Registry documentation for a full list of supported shell environment variables (e.g., authentication variables for [AWS](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#environment-variables), [Google Cloud Platform](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference#full-reference), and [Azure](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#argument-reference)). Environment variables can also [customize Terraform's behavior](/docs/cli/config/environment-variables.html). For example, `TF_LOG` enables detailed logs for debugging.
```

**Comment by laurapacilio** [comment]
Note that I'm in conversations with the google cloud provider team to make an update to the getting started guide credentials section that will have information about how to do this for users. So:
- I changed the GCP link back to the correct getting started guide section
- I will wait to merge this PR until that one gets put up and merged (they will do it for me!) 

I'll link the corresponding cloud provider PR when available.

**Comment by laurapacilio** [comment]
@im2nguyen @judithpatudith Update to provider docs has been merged, so I'm gonna push! 🥳

**Pull request merged**
